RadixWallet/Clients/FactorSourcesClient/FactorSourcesClient+Interface.swift

```diff
@@ -285,11 +285,17 @@ extension FactorSourcesClient {
 		onMnemonicExistsStrategy: ImportMnemonic.State.PersistStrategy.OnMnemonicExistsStrategy,
 		saveIntoProfile: Bool
 	) async throws -> DeviceFactorSource {
-		let isOlympiaCompatible = onDeviceMnemonicKind == .olympia
-
-		let factorSource: DeviceFactorSource = try isOlympiaCompatible
-			? .olympia(mnemonicWithPassphrase: mnemonicWithPassphrase)
-			: .babylon(mnemonicWithPassphrase: mnemonicWithPassphrase)
+		let factorSource: DeviceFactorSource = try { () throws -> DeviceFactorSource in
```

Comment:
Can't this be a normal switch expression?

Reply:
sure! I'm... somewhat torn between dropping the wrapping closures... but in this case it was fine, fixed!